doc/parameter_server.md

```diff
@@ -0,0 +1,144 @@
+# Parameter Server Design
+
+## Overview
+
+A typical parameter server(pserver) architecture contains three roles:
```

Comment:
We need an extra space here: `parameter server(pserver)` => `parameter server (pserver)`.

Reply:
Done